homeassistant/components/camera/logi_circle.py

```diff
@@ -0,0 +1,204 @@
+    vol.Required(ATTR_VALUE): cv.boolean
+})
+
+LOGI_CIRCLE_SERVICE_SNAPSHOT = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_FILENAME): cv.template
+})
+
+LOGI_CIRCLE_SERVICE_RECORD = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_FILENAME): cv.template,
+    vol.Required(ATTR_DURATION): cv.positive_int
+})
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Set up a Logi Circle Camera."""
+    devices = hass.data[DOMAIN]
+
+    cameras = []
+    for device in devices:
+        cameras.append(LogiCam(hass, device, config))
+
+    async_add_entities(cameras, True)
+
+    async def service_handler(service):
+        """Dispatch service calls to target entities."""
+        params = {key: value for key, value in service.data.items()
+                  if key != ATTR_ENTITY_ID}
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids:
+            target_devices = [dev for dev in cameras
+                              if dev.entity_id in entity_ids]
+        else:
+            target_devices = cameras
+
+        for target_device in target_devices:
+            if service.service == SERVICE_SET_CONFIG:
+                await target_device.set_config(**params)
+            if service.service == SERVICE_LIVESTREAM_SNAPSHOT:
+                await target_device.livestream_snapshot(**params)
+            if service.service == SERVICE_LIVESTREAM_RECORD:
+                await target_device.download_livestream(**params)
+
+    hass.services.async_register(
+        CAMERA_DOMAIN, SERVICE_SET_CONFIG, service_handler,
+        schema=LOGI_CIRCLE_SERVICE_SET_CONFIG)
+
+    hass.services.async_register(
+        CAMERA_DOMAIN, SERVICE_LIVESTREAM_SNAPSHOT, service_handler,
+        schema=LOGI_CIRCLE_SERVICE_SNAPSHOT)
+
+    hass.services.async_register(
+        CAMERA_DOMAIN, SERVICE_LIVESTREAM_RECORD, service_handler,
+        schema=LOGI_CIRCLE_SERVICE_RECORD)
+
+
+class LogiCam(Camera):
+    """An implementation of a Logi Circle camera."""
+
+    def __init__(self, hass, camera, device_info):
+        """Initialize Logi Circle camera."""
+        super(LogiCam, self).__init__()
+        self._camera = camera
+        self._name = self._camera.name
+
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Logi Circle camera's support turning on and off ("soft" switch)."""
+        return SUPPORT_ON_OFF
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            'battery_saving_mode': ('on' if self._camera.battery_saving
+                                    else 'off'),
+            'firmware': self._camera.firmware,
```

Comment:
Understood. Based on this, I've removed the following attributes:

* Firmware
* Generation
* MAC address
* Model
* Mount
* Plan
* Speaker volume (is related to state, but duplicates a sensor)
* Timezone

I don't think these attributes add that much value - are you comfortable if I take a look at potentially adding config entries in a future PR rather than attack it now?